packages/slice-machine/lib/mock/Slice.ts

```diff
@@ -51,20 +51,38 @@ export function buildSliceMockConfig(
 
   return model.variations.map((v) => {
     return {
+      type: "SharedSlice",
       variation: v.id,
       variations: variationConfigs,
     };
   });
 }
 
 export default function MockSlice(
-  smModel: SliceSM,
-  legacyMockConfig: Record<string, Record<string, Record<string, unknown>>> // not sure about this one.
+  sliceModel: SharedSlice,
+  legacyMockConfig: Record<string, Record<string, Record<string, unknown>>>, // not sure about this one.
+  previousMocks?: ComponentMocks | null | undefined,
+  sliceDiff?: SliceDiff | undefined
 ): ComponentMocks {
-  const model = Slices.fromSM(smModel);
-
-  const sliceMockConfig = buildSliceMockConfig(model, legacyMockConfig);
+  const sliceMockConfig = buildSliceMockConfig(sliceModel, legacyMockConfig);
   return sliceMockConfig.map((sc) => {
-    return SharedSliceMock.generate(model, sc);
+    const variationMock = previousMocks?.find(
+      (m) => m.variation === sc.variation
+    );
+    if (sliceDiff) {
+      if (!variationMock) return SharedSliceMock.generate(sliceModel, sc);
+
+      const patched = SharedSliceMock.patch(
+        sliceDiff,
+        sliceModel,
+        variationMock,
+        sc
+      );
+      if (!patched.ok || !patched.result)
+        return SharedSliceMock.generate(sliceModel, sc);
+      return patched.result;
+    }
+
+    return SharedSliceMock.generate(sliceModel, sc);
```

Comment:
would it make more sense (less nesting) to have a fefw of these conditions as an early exit?
```
if(!sliceDiff) return SharedSliceMock.generate(sliceModel, sc);
const variationMock = ...
if(!variationMock) return  SharedSliceMock.generate(sliceModel, sc);
const patched = ...
if(patchedNotGood) return SharedSliceMock.generate(sliceModel, sc);
return patched.result
```

Reply:
Surprised this didn't break the tests :)